### PR TITLE
Adds an optional, second hotkey modifier

### LINKED ISSUE #170: Multiple keys for hotKey combination
If I understand correctly there is no possibility to use multiple keys for hotKey combination such as Shift+AltGr+Space as it was in the past? Could we make that possible again?

### LINKED ISSUE #416: More hotkeys
It would be great to be able to define a specific hotkey outside the current selection.

### LINKED ISSUE #453: Allow a third key for hotkey shortcut
Currently two keys are required to construct the shortcut to open the ueli search window. As many keys are supported, it may introduce hotkey conflicts with Windows (probably MacOS too) and other programs. Allowing a third key would avoid those conflicts.

### PR DESCRIPTION
**Pull request by tvollstaedt**
This PR add's an optional, second hotkey modifier to use up to three keys as a global ueli hotkey.

Also adds "A-Z" key options for hotkey, as they are supported by electron. 
Also fixes a non-working "Ctrl" bug on Windows (see https://www.electronjs.org/docs/api/accelerator#platform-notice).

Closes #416 
Closes #453 
Closes #170

### CODE REVIEW

**Comment by jonathanantoine** [comment]
Any update on this PR ? @oliverschwendener  :) ? Would love to see this happen !

**Comment by istatv** [comment]
> Any update on this PR ? @oliverschwendener :) ? Would love to see this happen !

Unfortunately, PR's are not actively maintained in this project, we'll probably have to start a fork to really make use of community contributions.

**Comment by tvollstaedt** [comment]
> Any update on this PR ? @oliverschwendener :) ? Would love to see this happen !

Unfortunately, PR's are not actively maintained in this project, we'll probably have to start a fork to really make use of community contributions.

**Comment by oliverschwendener** [comment]
Yea sorry I try to keep up but I hardly find time to review all PRs 😅

**Comment by tvollstaedt** [comment]
> Yea sorry I try to keep up but I hardly find time to review all PRs 😅

Don't worry totally understandable. But you could add some collaborators to the project and share the work :-)

**Comment by oliverschwendener** [comment]
> Don't worry totally understandable. But you could add some collaborators to the project and share the work :-)

Until now I didn't see anyone contributing to this project yet, who I would trust to maintain this project and making sure that it does not turn into a unmaintainable piece of software. On the other hand, this project is open source and can be forked and changed as much as you like ;)

**Comment by oliverschwendener** [comment]
@tvollstaedt Can you please remove all changes that are only related to formatting? It would make this change much smaller and easiert to review.

**Comment by tvollstaedt** [comment]
> @tvollstaedt Can you please remove all changes that are only related to formatting? It would make this change much smaller and easiert to review.

Done.

**Comment by centur** [comment]
hope this will be merged soon, as after I found Ueli - I want to switch from Wox (ueli is much more active project atm) but keybindings are very very limited. Looking forward this PR to be merged and released in the wild :) 
Really appreciate this great launcher!

**Comment by oliverschwendener** [comment]
Currently I have set Alt+Space as my keyboard shortcut. When I checkout this branch this is what I see in the settings:

image

In my opinion this is not very nice. I suggest that we either provide a way to migrate the old settings to the new structure (`modifier` renamed to `firstModifier`) or we don't rename `modifier` and just add `secondModifier`. Additionally I think we should also make it clear in the UI that both modifiers are optional. With your solution it looks like they are both required.

**Comment by tvollstaedt** [comment]
Yeah your'e right, if you start without an existing configuration, it will look like this:

image

So I guess I will need to add some config migration logic to not screw upgrading users.

File: src/common/global-hot-key/global-hot-key.ts [hunk 1]
```diff
@@ -3,5 +3,6 @@ import { GlobalHotKeyKey } from "./global-hot-key-key";
 
 export interface GlobalHotKey {
     key: GlobalHotKeyKey;
-    modifier?: GlobalHotKeyModifier;
+    firstModifier?: GlobalHotKeyModifier;
```
Comment (oliverschwendener) [inline src/common/global-hot-key/global-hot-key.ts:6]: If we wouldn't rename `modifier` to `firstModifier` we wouldn't need any migration. Why don't we just add `secondModifier` ?

**Comment by oliverschwendener** [comment]
I guess this was not intended?

image

File: src/common/global-hot-key/global-hot-key.ts (hunk 1 above)
Comment (tvollstaedt) [reply src/common/global-hot-key/global-hot-key.ts]: Alright, fair enough. I've renamed `firstModifier` and added small fallback logic for missing `secondModifier`.

**Comment by tvollstaedt** [comment]
> I guess this was not intended?
> 
> image

Nope, sorry.

File: src/common/global-hot-key/global-hot-key-helpers.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import { GlobalHotKeyKey } from "./global-hot-key-key";
 import { GlobalHotKeyModifier } from "./global-hot-key-modifier";
 
 export function isValidHotKey(hotKey: GlobalHotKey) {
-    return Object.keys(GlobalHotKeyKey).find((k) => k === hotKey.key) !== undefined
-        && (hotKey.modifier ? Object.keys(GlobalHotKeyModifier).find((m) => m === hotKey.modifier) : true);
+    return Object.values(GlobalHotKeyModifier).includes(hotKey.modifier!) &&
+        Object.values(GlobalHotKeyModifier).includes(hotKey.secondModifier!) &&
+        Object.values(GlobalHotKeyKey).includes(hotKey.key || "");
```
Comment (oliverschwendener) [inline src/common/global-hot-key/global-hot-key-helpers.ts:8]: Why is the `|| ""` needed?

Comment (oliverschwendener) [reply src/common/global-hot-key/global-hot-key-helpers.ts:8]: @tvollstaedt ?

Comment (tvollstaedt) [reply src/common/global-hot-key/global-hot-key-helpers.ts:8]: I don't remember anymore why I did this. `key` is not optional, so it should be always there.

```suggestion
        Object.values(GlobalHotKeyKey).includes(hotKey.key);
```

**Comment by oliverschwendener** [comment]
Looks good to me now, thanks for your efforts and sorry for the long wait! 😅

**Pull request merged**